### DISCUSSION TITLE
Render object with coated thin dielectric BRDF

### DISCUSSION THREAD

**Discussion by duongnlt** [mitsuba-renderer/mitsuba3#1248] [Q&A]
Hi team,

My question is whether there is a way to render an object using BRDF data from the MERL dataset, while also applying a thin dielectric BRDF coating to that object?

Thank you!

**Comment by njroussel** [comment]
Hi @duongnlt 
I can't think of an easy/hacky way of doing this, I think you'll have to merge some plugins together to get similar functionality.

**Reply by duongnlt** [reply]
Hi @njroussel,

I think what I want to achieve is similar to the Smooth Dielectric Coating BSDF in Mitsuba 0.6. Is it feasible to develop a custom plugin for this in Python? If so, could you offer any suggestions?

**Reply by njroussel** [reply]
I wasn't aware of that plugin, I just looked it up.

I think that's entirely possible. Nothing is stopping you form defining a nested BSDF and using it in some "wrapper" BSDF in Mitsuba 3.
